reflection.md

```diff
@@ -15,4 +26,18 @@ Write a few brief paragraphs that discuss:
    you implement it _or_ what features of Scala would prevent you from
    implementing it? (You don't have to write code for this part. You could say
    "I would use literal extension to..." or "Scala's rules for valid
-   identifiers prevent...")
\ No newline at end of file
+   identifiers prevent...")
+
+   I think this language is very well designed.  It is easy to say what a pattern
+   is and to combine multiple patterns. It is also easy to match a pattern to
+   a string, which is critical.  
+
+   It is difficult/impossible to say more than whether a string matches a pattern.
+   Lots of modern regex implementations I believe offer ways to have capture
+   groups and match certain parts of strings and extract information you want
+   from strings using the pattern you specify.  This could be a nice extension
+   of this project.
+
```

Comment:
Good point.  Capturing expressions and replacing them would be a great addition to this language.